components/post_view/post_message_preview/post_message_preview.tsx

```diff
@@ -0,0 +1,92 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+
+import {FormattedMessage} from 'react-intl';
+
+import {Post, PostPreviewMetadata} from 'mattermost-redux/types/posts';
+import UserProfileComponent from 'components/user_profile';
+import {UserProfile} from 'mattermost-redux/types/users';
+import Avatar from 'components/widgets/users/avatar';
+import * as Utils from 'utils/utils';
+import PostMessageView from 'components/post_view/post_message_view';
+
+import Timestamp from 'components/timestamp';
+import PostAttachmentContainer from '../post_attachment_container/post_attachment_container';
+
+export type Props = {
+    user?: UserProfile;
+    post?: Post;
+    metadata: PostPreviewMetadata;
+};
+
+const PostMessagePreview = (props: Props) => {
+    const {user, metadata, post} = props;
+    if (!post || !user) {
+        return null;
+    }
+    return (
+        <PostAttachmentContainer
+            className='permalink'
+            link={`/${metadata.team_name}/pl/${metadata.post_id}`}
+        >
+            <div className='post-preview'>
+                <div className='post-preview__header'>
+                    <div className='col col__name'>
+                        <div className='post__img'>
+                            <span className='profile-icon'>
+                                <Avatar
+                                    username={user?.username}
```

Comment:
`username` (and a few lines down `id`) are both required fields of `UserProfile` so the optional chaining operator will compile to unnecessary JavaScript conditions. 0/5 whether it affects performance and/or JS compilation size enough to care.